readme.md

```diff
@@ -0,0 +1,161 @@
+# Remark for Visual Studio Code
+
+[![Build][build-badge]][build]
+[![Coverage][coverage-badge]][coverage]
+[![Downloads][downloads-badge]][downloads]
+[![Sponsors][sponsors-badge]][collective]
+[![Backers][backers-badge]][collective]
+[![Chat][chat-badge]][chat]
+
+**[remark][]** is a tool that transforms markdown with plugins.
+These plugins can inspect and change your markup.
+You can use remark on the server, the client, CLIs, deno, etc.
+This plugin for Visual Studio Code lets you format and lint markdown files using
+remark.
+
+## Contents
+
+*   [When should I use this?](#when-should-i-use-this)
+*   [Use](#use)
+    *   [Formatting](#formatting)
+*   [Plugins](#plugins)
+*   [Syntax](#syntax)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## When should I use this?
+
+If your project is checked using [`remark-cli`][remark-cli] and you use Visual
+Studio Code to edit markdown files, it’s recommended to use this extension.
+
+## Use
+
+To start using this for a project, [`remark-cli`][remark-cli] needs to be
+configured first.
+For example, install `remark-cli` and `remark-preset-lint-recommended`:
+
+```sh
+npm install remark-cli remark-preset-lint-recommended
+```
+
+Now create a file named `.remarkrc.json` with the following content:
+
+```json
+{
+  "plugins": [["preset-lint-recommended"]]
+}
+```
+
+Now any syntax which doesn’t comply with `remark-preset-lint-recommended` will
+be reported in the *Problems* pane in your editor.
+
+### Formatting
+
+This extension can be used as a formatter for markdown files.
+
+To format a file, first press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>.
```

Comment:
is formatted a bit verbose, why not one `kbd` with the plusses inside?

Reply:
I didn’t even know if your suggestion is valid, so I googled `html kbd`. It turnes out some people use it, but more people favor the format I’m using here.

It also makes more sense to me. You don’t press a single key which is labeled <kbd>Ctrl + Shift + P</kbd>. You press 3 keyboard keys: <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd> (maybe some spacing would be nice though)

Reply:
Interesting!!

> ou don’t press a single key which is labeled Ctrl + Shift + P. You press 3 keyboard keys: Ctrl+Shift+P (maybe some spacing would be nice though)

But you have to press them at the same time. Your latter example to me indicated something done in sequence rather than at the same time.

>  but more people favor the format I’m using here.

Could you enlighten me on what you found, that indicates this?

---

The first example two examples in the HTML spec is way more complex than what we both do: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-kbd-element.
The third example, seems to be more in line with what I’d suggest. Prefixed with “Such precision isn't necessary; the following is equally fine”

Reply:
Interestingly the readme of this project currently uses my style: https://github.com/remarkjs/vscode-remark/blob/4830030f626cc24bff3eb23f8dd99c42fa69af09/README.md?plain=1#L20-L21.
I do find a lot of examples of your style though.

Reply:
One more reason in favor of my style: the plusses are weird in your style. They’re not natural language. Why not use `and`? Or commas? Makes more sense to me when they’re inside the `kbd`

Reply:
Some examples which use my style are the [kbd docs on mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/kbd), [kbd docs on w3schools](https://www.w3schools.com/tags/tag_kbd.asp) and [Wikipedia article on Ctrl-Alt-Delete](https://en.wikipedia.org/wiki/Control-Alt-Delete). IMHO this also makes sense, as it’s rendered to look similar to a key on a keyboard.

I’m also pretty sure this is the style I see most of the time, although it’s hard to find examples when actually looking for them.

From https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-kbd-element:

> When the `kbd` element is nested inside another `kbd` element, it represents an actual key or other single unit of input as appropriate for the input mechanism.
> […]
> > <p>To make George eat an apple, press <kbd><kbd>Shift</kbd>+<kbd>F3</kbd></kbd></p>

So then I’d say it should be:

<kbd><kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd></kbd>

This also looks fine with me.

Reply:
Yeah that last one looks a bit verbose to me (which the intro for the 3rd example also seems to implicated). I'm fine with "your style" 👍